### PR TITLE
Add isCarrierSpecificForRegion() API in ShortNumberInfo class.

### PR DESCRIPTION
**Pull request by penmetsaa**
This determines whether the provided short code is carrier-specific or not when dialed from the given region.

b/24891929
cl/142241335

---
This change is [Reviewable](https://reviewable.io/reviews/googlei18n/libphonenumber/1504)

### CODE REVIEW

File: cpp/src/phonenumbers/shortnumberinfo.h [hunk 1]
```diff
@@ -149,6 +149,16 @@ class ShortNumberInfo {
...
 
+  // Given a valid short number, determines whether it is carrier-specific when
+  // dialed from the given region (however, nothing is implied about its
+  // validity). If it is important that the number is valid, then its validity
+  // must first be checked using IsValidShortNumber or
+  // IsValidShortNumberForRegion. Returns false if the number doesn't match the
+  // region provided.
+  bool IsCarrierSpecificForRegion(
+      const PhoneNumber& number,
+      const string& region_code) const;
```
Comment (keghani) [inline cpp/src/phonenumbers/shortnumberinfo.h]: please make region_dialing_from (to be consistent with other files)

Comment (penmetsaa) [inline cpp/src/phonenumbers/shortnumberinfo.h]: Done Keghani.

Comment (keghani) [inline cpp/src/phonenumbers/shortnumberinfo.h]: Not Done?

**Comment by penmetsaa** [comment]
Hi Keghani,
Made suggested changes. Please review.

File: cpp/src/phonenumbers/shortnumberinfo.h [hunk 2]
```diff
@@ -144,11 +144,24 @@ class ShortNumberInfo {
...
+  // Given a valid short number, determines whether it is carrier-specific when
+  // dialed from the given region (however, nothing is implied about its
+  // validity). Carrier-specific numbers may connect to a different end-point,
+  // or not connect at all, depending on the user's carrier. If it is important
+  // that the number is valid, then its validity must first be checked using
+  // IsValidShortNumber or IsValidShortNumberForRegion. Returns false if the
+  // number doesn't match the region provided.
+  bool IsCarrierSpecificForRegion(
+      const PhoneNumber& number,
+      i18n_identifiers::RegionCode region_dialing_from) const;
```
Comment (keghani) [inline cpp/src/phonenumbers/shortnumberinfo.h]: string&

Comment (keghani) [inline cpp/src/phonenumbers/shortnumberinfo.h]: (and const)

**Comment by penmetsaa** [comment]
Oh sorry Keghani. I have copied method signature also.
Corrected it now. PTAL.

**Comment by penmetsaa** [comment]
Hi Keghani,
Forgot to mention. All CPP and Java tests are passing now.

**Pull request merged**